src/layout/stack.scss

```diff
@@ -0,0 +1,286 @@
+// Stack layout helper component
+
+.Stack {
+  
+  // A Stack component lays elements horizontally or vertically on the page.
+  //
+  // Stack is a simple abstraction of CSS' Flexbox. Use it to structure elements
+  // that are visually grouped, following the same direction.
+  //
+  // Markup structure
+  // =========
+  //
+  // .Stack
+  // ├─ &.Stack--dir-[ inline | block ]-[ whenRegular | whenNarrow ]
+  // ├─ &.Stack--gap-[ none | condensed | normal | spacious ]-[ whenRegular | whenNarrow ]
+  // ├─ &.Stack--align-[ start | center | end | baseline ]-[ whenRegular | whenNarrow ]
+  // ├─ &.Stack--alignWrap-[ start | center | end | distribute | distributeEvenly ]-[ whenRegular | whenNarrow ]
+  // ├─ &.Stack--spread-[ start | center | end | distribute | distributeEvenly ]-[ whenRegular | whenNarrow ]
+  // ├─ &.Stack--wrap-[ whenRegular | whenNarrow ]
+  // ├─ &.Stack--nowrap-[ whenRegular | whenNarrow ]
+  // ├─ &.Stack--showDividers-[ whenRegular | whenNarrow ]
+  // │
+  // ├─ .Stack-divider
+
+  --Stack-gap: var(--primer-stack-gap-normal, 16px);
+  --Stack-gap-whenNarrow: var(--Stack-gap);
+  --Stack-divider-color: var(--color-border-default);
+
+  display: flex;
+  flex-flow: column;
+  flex-wrap: nowrap;
+  align-items: stretch;
+  gap: var(--Stack-gap);
+  // min-block-size: fit-content;
+  // min-inline-size: fit-content;
+
+  @media ($viewport-narrow) {
+    gap: var(--Stack-gap-whenNarrow);
+  }
+}
+
+/* Direction */
+
+@media ($viewport-regular) {
+  .Stack--dir-inline-whenRegular {
+    flex-flow: row;
+  }
+
+  .Stack--dir-block-whenRegular {
+    flex-flow: column;
+  }
+}
+
+@media ($viewport-narrow) {
+  .Stack--dir-inline-whenNarrow {
+    flex-flow: row;
+  }
+
+  .Stack--dir-block-whenNarrow {
+    flex-flow: column;
+  }
+}
+
+/* Gap */
+
+@media ($viewport-regular) {
+  .Stack--gap-none-whenRegular {
```

Comment:
Do we need the `-none-when{ViewportRange}` class names if we're already using media queries?

Reply:
We do, since we're also optimizing the stylesheet to accept default states without including all modifier classes. That should make it easier to consume the component if it's not generated by Primer ViewComponent or Primer React directly (it also outputs a cleaner markup 😇).

For example:

```html
<div class="Stack">
  ...
</div>
```

☝️ is a valid markup, which sets direction to `block`, gap to `normal`.

If a user wants set `gap` to `none`, that'll be passed as a modifier: `Stack--gap-none-{viewportRange}`.

Note that viewport range information is needed so the user can control in the markup in which viewport the property will be applied.

Some valid scenarios:

```html
<!-- gap set to `normal` on regular, and to `condensed` on narrow:
<div class="Stack Stack--gap-condensed-whenNarrow">
  ...
</div>

<!-- gap set to `none` on regular, and to `normal` on narrow:
<div class="Stack Stack--gap-none-whenRegular">
  ...
</div>
```

Reply:
That was a very silly mistake on my part. You're absolutely right.